### PR TITLE
Seccon 12666 -Make search results retrieval use translated query instead of reasembling it - this makes the automated tests cover the code flow

### PR DESCRIPTION
**Pull request by Guy-Galil**

### CODE REVIEW

**Comment by mdazam1942** [comment]
@Guy-Galil would you please give a more descriptive title of the PR instead of JIRA issue number?

File: stix_shifter_modules/guardium/stix_transmission/api_client.py [hunk 1]
```diff
@@ -102,12 +102,10 @@ def get_search_results(self, search_id,  index_from=None, fetch_size=None):
             indx = int(index_from)+1
             fsize = int(fetch_size)+1
             if "reportName" in self.query:
-                response = self.client.handle_report(self.query["reportName"], self.query["reportParameter"], indx, fsize)
-                respObj = ResponseWrapper(response)
+                response = self.client.handle_report(self.query, indx, fsize)
```
Comment (mdazam1942) [inline stix_shifter_modules/guardium/stix_transmission/api_client.py:105]: why the `reportName` or `reportParameter` not used to make the api call anymore? If the they are not applicable anymore then we should also update the query construction.

Comment (Guy-Galil) [reply stix_shifter_modules/guardium/stix_transmission/api_client.py:105]: reportName and reportParameter are put into the query in the query construction step, I removed redundant code that was reassembling the query again before execution

File: stix_shifter_modules/guardium/stix_transmission/guard_utils.py [hunk 2]
```diff
@@ -108,15 +108,16 @@ def validate_response(self, p_response, prefix, abort=False):
...
         results = ""
         #context().logger.debug('-------------------  ' + report_name + ' ----------------------')
-        params_set = {"reportName":"{0}".format(report_name), "indexFrom": "{0}".format(index_from), "fetchSize": "{0}".format(fetch_size), 
-        "reportParameter":params, "inputTZ":"UTC"}
-        json_dump = json.dumps(params_set)
+        params["fetchSize"] = int(fetch_size)
+        params["index_from"]=int(index_from)
+        params["inputTZ"]="UTC"
+        json_dump = json.dumps(params)
         rest_data = str(json.loads(json_dump))
```
Comment (mdazam1942) [inline stix_shifter_modules/guardium/stix_transmission/guard_utils.py:121]:
```
       json_dump = json.dumps(params)
         rest_data = str(json.loads(json_dump))
```
I don't think you need to dump the dictionary object and then load and again cast to string. Seems like some redundant operations. You may be able to use the `params` in the request method.

File: stix_shifter_modules/guardium/stix_transmission/guard_utils.py [hunk 3]
```diff
@@ -137,26 +138,19 @@ def handle_report(self, report_name, params, index_from, fetch_size):
...
-                else: 
-                    params_set["filters"] = ""
-        all_params = {**params_set, **params}
-        json_dump = json.dumps(all_params)
+        params["fetchSize"] = int(fetch_size-1)
+        params["firstPosition"]=int(index_from-1)
+        params["inputTZ"]="UTC"
+
+        json_dump = json.dumps(params)
         rest_data = str(json.loads(json_dump))
```
Comment (mdazam1942) [inline stix_shifter_modules/guardium/stix_transmission/guard_utils.py:154]: similar to https://github.com/opencybersecurityalliance/stix-shifter/pull/577/files#r608212762

File: stix_shifter_modules/guardium/stix_transmission/guard_utils.py (hunk 2 above)
Comment (Guy-Galil) [reply stix_shifter_modules/guardium/stix_transmission/guard_utils.py]: reportName and reportParameter are put into the query in the query construction step, I removed redundant code that was reassembling the query again before execution

Comment (Guy-Galil) [reply stix_shifter_modules/guardium/stix_transmission/guard_utils.py]: Removed redundant json conversion

File: stix_shifter_modules/guardium/stix_transmission/guard_utils.py (hunk 3 above)
Comment (Guy-Galil) [reply stix_shifter_modules/guardium/stix_transmission/guard_utils.py]: Removed redundant json conversion

**Review by mdazam1942 — APPROVED** [review]

**Review by mdazam1942 — APPROVED** [review]

**Pull request merged**